src/controllers/edgeconnect/controller.go

```diff
@@ -66,6 +67,13 @@ func (controller *Controller) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{RequeueAfter: errorUpdateInterval}, nil
 	}
 
+	log.Info("updating version info", "name", request.Name, "namespace", request.Namespace)
+	versionReconciler := version.NewReconciler(edgeConnect, controller.apiReader, timeprovider.New().Freeze())
+	if err = versionReconciler.Reconcile(ctx); err != nil {
```

Comment:
nitpick: `timeprovider.Freeze()` is mainly meant for testing, I don't think it causes problems here, but I also see no benefit. 🤷